scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -2141,7 +2141,7 @@ class ScalametaParser(input: Input)(implicit dialect: Dialect) { parser =>
           canApply = false
           Success(autoPos {
             next()
-            template(enumCaseAllowed = false, secondaryConstructorAllowed = false) match {
+            template(OwnedByTrait) match {
```

Comment:
Why use trait here? Can it not be a class also?

Reply:
Actually nevermind, makes sense!

Reply:
i think because class allowed secondary constructor 🙂